vertexai/token-count/tokencount_multimodal.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// token-count shows an example of determining how many tokens correspond to
+// a given prompt string
+package tokencount
+
+// [START generativeaionvertexai_gemini_token_count_multimodal]
+import (
+	"context"
+	"fmt"
+	"io"
+	"mime"
+	"path/filepath"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+// countTokensMultimodal returns the number of tokens for this prompt.
+//
+// video is a Google Cloud Storage path starting with "gs://"
+func countTokensMultimodal(w io.Writer, prompt, video, projectID, location, modelName string) error {
+	// prompt := "Provide a description of the video."
+	// video := "gs://cloud-samples-data/generative-ai/video/pixel8.mp4"
+	// location := "us-central1"
+	// modelName := "gemini-1.5-flash-001"
+
+	ctx := context.Background()
+
+	client, err := genai.NewClient(ctx, projectID, location)
+	if err != nil {
+		return fmt.Errorf("unable to create client: %w", err)
+	}
+	defer client.Close()
+
+	model := client.GenerativeModel(modelName)
+
+	part1 := genai.Text(prompt)
+
+	// Given a video file URL, prepare video file as genai.Part
+	part2 := genai.FileData{
+		MIMEType: mime.TypeByExtension(filepath.Ext(video)),
+		FileURI:  video,
+	}
+
+	resp, err := model.CountTokens(ctx, part1, part2)
```

Comment:
It would be useful to have comments here to explain what's going on. This seems to be demonstrating two different ways of getting a token count - one by calling `CountTokens` and another by examining the `UsageMetadata` returned by `GenerateContent`. The comments should say that